Improvement/test namnx228

namnx228 opened the pull request.

maelk commented on vm-setup/roles/libvirt/tasks/install_setup_tasks.yml:
```diff
@@ -1,29 +1,7 @@
+- name: Install libvirt dependencies
+  block:
+    - include_tasks: install_package_tasks.yml
+  when: libvirt_action == "install_package"
```
Is this needed if it runs in 01 ? and starting libvirtd could be considered as an installation step

maelk commented on vm-setup/roles/libvirt/tasks/main.yml:
```diff
@@ -1,10 +1,14 @@
     - include_tasks: vm_setup_tasks.yml
       when: vm_platform == "libvirt"
-  when: libvirt_action == "setup"
+  when: libvirt_action == "setup" 
```
unneeded space

maelk reviewed (commented):
Let's discuss about the use of roles

maelk commented on vm-setup/install-package-playbook.yml:
```diff
@@ -0,0 +1,15 @@
+  connection: local
+  gather_facts: true
+  vars_files:
+    - vars/input.yml
```
please use more explicit file naming

maelk commented on vm-setup/install-package-playbook.yml:
```diff
@@ -0,0 +1,15 @@
+    - import_role:
+        name: packages_installation
+  vars: 
+    install: "{{ input.packages_installation }}"
```
you can remove one level here

maelk commented on vm-setup/roles/common/tasks/main.yml:
```diff
@@ -1,70 +1,72 @@
-                - domain: "apps.{{ cluster_domain }}"
-                  addr: "127.0.0.1"
+- name: Setup VMs
+  when: libvirt_action != "install_package"
```
remove this condition, it is not needed

maelk commented on vm-setup/roles/packages_installation/tasks/main.yml:
```diff
@@ -0,0 +1,36 @@
+        - name: Install packages on rhel8
+          when: ansible_lsb.major_release|int == 8
+          package:
+             name: "{{ install.centos.package + install.centos.libvirt.rhel7 }}"
```
rhel8 here

maelk commented on vm-setup/vars/input.yml:
```diff
@@ -0,0 +1,96 @@
+input:
```
you can remove the input

nordixinfra commented:
Can one of the admins verify this patch?

namnx228 committed:
Move all download code into 01_install_requirement.sh

New ansible playbook to download packages

Install redhat-lsb-core  using yum to make  `ansible_lsb` available

maelk commented on vm-setup/roles/common/tasks/main.yml:
```diff
@@ -1,70 +1,71 @@
-                - domain: "apps.{{ cluster_domain }}"
-                  addr: "127.0.0.1"
+- name: Setup VMs
+  block: 
```
why a block here ?

maelk replied:
This file should not be touched, right?

maelk commented on vm-setup/vars/packages_installation.yml:
```diff
@@ -0,0 +1,95 @@
+packages_installation:
+  ubuntu:
+    package:
```
packages

maelk commented on vm-setup/vars/packages_installation.yml:
```diff
@@ -0,0 +1,95 @@
+     - setuptools 
+     - libvirt-python 
+  centos:
+    package:
```
packages

maelk commented on vm-setup/roles/packages_installation/tasks/main.yml:
```diff
@@ -0,0 +1,36 @@
+  block:
+    - name: Install packages using standard package manager
+      package:
+        name: "{{ package.ubuntu.package +  package.ubuntu.libvirt }}"
```
packages (*3)

maelk commented on vm-setup/roles/packages_installation/tasks/main.yml:
```diff
@@ -0,0 +1,36 @@
+      become: true
+    - name: Install packages using pip
+      pip:
+        name: "{{ package.ubuntu.pip }}"
```
same and further in the file

maelk commented on vm-setup/install-package-playbook.yml:
```diff
@@ -0,0 +1,15 @@
+    - import_role:
+        name: packages_installation
+  vars: 
+    package: "{{ packages_installation }}"
```
packages

namnx228 committed:
Fix typos

maelk commented:
we will not continue further in this direction for now
